proxy/tests/support/server.rs

```diff
@@ -30,6 +30,13 @@ pub struct Server {
 pub struct Listening {
     pub addr: SocketAddr,
     pub(super) shutdown: Shutdown,
+    pub(super) conn_count: Arc<Mutex<usize>>,
```

Comment:
Prefer an `AtomicUsize` in almost any case where you want a `Mutex<usize>`.

Reply:
Done! 954d64193a6e1025d95a633cfe988b9551fef503